ui/ui-components/components/session/SignIn.tsx

```diff
@@ -52,50 +103,133 @@ export default function SignInForm(props) {
+          >
+            <Form.Group>
+              <Form.Label>Email Address</Form.Label>
+              <Form.Control
+                disabled={loading}
+                autoFocus
+                required
+                name="email"
+                type="email"
+                placeholder="Email Address"
+                ref={register}
+              />
+              <Form.Control.Feedback type="invalid">
+                Email is required
+              </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group>
+              <Form.Label>Password</Form.Label>
+              <Form.Control
+                required
+                disabled={loading}
+                name="password"
+                type="password"
+                placeholder="Password"
+                ref={register}
+              />
+              <Form.Control.Feedback type="invalid">
+                A password is required
+              </Form.Control.Feedback>
+            </Form.Group>
+
+            <LoadingButton loading={loading} variant="primary" type="submit">
+              Sign In
+            </LoadingButton>
+          </Form>
+        </Col>
+
+        <Col>
+          <h2>Sign in with oAuth</h2>
+
+          {loadingOauthProviders ? <Loader /> : null}
+
+          <br />
+
+          {loadingOauthProviders === false && providers.length === 0 ? (
+            <p>Could not load oAuth providers</p>
+          ) : null}
+
+          <div className="d-flex justify-content-center">
+            <ButtonGroup vertical style={{ width: "100%" }}>
+              {providers.map((provider) => (
+                <Fragment key={`provider-${provider.name}`}>
+                  <LoadingButton
+                    disabled={loading}
+                    loading={loadingOauthProviders}
+                    variant="outline-primary"
+                    onClick={() => startOauthLogin(provider.name, "user")}
+                  >
+                    {provider.description}
+                    <img
+                      style={{ margin: 10, height: 40, width: 40 }}
+                      src={provider.icon}
+                    />
+                  </LoadingButton>
+                </Fragment>
+              ))}
+            </ButtonGroup>
+          </div>
+        </Col>
+      </Row>
+
+      <Modal show={showModal} onHide={() => setShowModal(false)}>
+        {oAuthRequest ? (
+          <>
+            <Modal.Header closeButton>
+              <Modal.Title>Select Identity</Modal.Title>
+            </Modal.Header>
+            <Modal.Body>
+              <ButtonGroup vertical style={{ width: "100%" }}>
+                {oAuthRequest.identities.map((identity, idx) => (
+                  <Button
+                    variant="outline-primary"
+                    key={`identity-${idx}`}
+                    onClick={() => {
+                      setShowModal(false);
+                      onSubmit(identity, "requestId");
+                    }}
```

Comment:
I wonder if there should be a confirmation step here that this is definitely the identity you want to use. (The functionality should be select the identity then confirm)

Reply:
I didn't want add any more clicks that we already had... 
If you pick the wrong identity, you can just do the oauth flow again, so you aren't locked out at least